src/libs/ck-libs/NDMeshStreamer/NDMeshStreamer.ci

```diff
@@ -1,17 +1,18 @@
 module NDMeshStreamer {
   extern module completion;
-
   include "DataItemTypes.h";
 
-  template<class dtype>
-  message MeshStreamerMessage {
+  message MeshStreamerMessageV {
     int destinationPes[];
-    dtype dataItems[];
+    int sourcePes[];
```

Comment:
Does this mean that TRAM messages are larger now for even fixed sized messages? Or do those use a different message?

Reply:
Yes, but note that the dtype in the old code actually would be something like a tuple<dtype, CkArrayIndex, int> in the new code, so the only extra item that is actually being added is the offset array.

Reply:
Yeah but I still think that should be avoided since it needlessly bloats the amount of bytes sent over the network. I see two optimization possibilities here:
1. Add a flag to the message structure whose value is either the fixed contribution size (which tells you to not bother with 'offsets') or some special value saying they're variable-sized.
2. Shrink the type of 'offsets' from 'int' to 'uint16_t' (or whatever integral type is the smallest capable one), since you know the buffer is never larger than the compile-time constant 'tramBufferSize', IIRC.